modules/messages/client/components/UnreadCount.component.js

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { watch as watchUnreadCount } from '@/modules/messages/client/services/unread-message-count.client.service';
+
+export default function UnreadCount() {
+  const { t } = useTranslation('messages');
+
+  const [count, setCount] = useState(null);
+
+  useEffect(() => watchUnreadCount(setCount), []);
```

Comment:
Yes I quite liked how that pattern turned out :)

Passing `setCount` to the service is just the same as doing it more verbosely, so, a full verbose version of it would be:

```js
const [count, setCount] = useState(null);

useEffect(() => {
  const unwatch = watchUnreadCount(count => {
    setCount(count);
  })
  return unwatch;
}, []);
```

In that more verbose one we're _effectively_ still passing `setCount` to the service, just wrapped up inside the anonymous arrow function that we pass it, but it's still there.

Arguably it's _less_ clear how I have it and a bit more _magic_, so open for opinions!